app/src/ui/changes/changes-list.tsx

```diff
@@ -428,7 +429,11 @@ export class ChangesList extends React.Component<
           coAuthors={this.props.coAuthors}
           placeholder={
             singleFileCommit
-              ? `Update ${filesSelected[0].path}`
+              ? filesSelected[0].status === AppFileStatus.New
```

Comment:
This feels like it could be lifted out to it's own function. I noodled on this version to illustrate how it might look:

```diff
diff --git a/app/src/ui/changes/changes-list.tsx b/app/src/ui/changes/changes-list.tsx
index 33ed077bd..4747f7828 100644
--- a/app/src/ui/changes/changes-list.tsx
+++ b/app/src/ui/changes/changes-list.tsx
@@ -31,7 +31,7 @@ import { IAutocompletionProvider } from '../autocompletion'
 import { showContextualMenu } from '../main-process-proxy'
 import { arrayEquals } from '../../lib/equality'
 import { clipboard } from 'electron'
-import * as path from 'path'
+import { parse } from 'path'
 
 const RowHeight = 29
 
@@ -378,6 +378,32 @@ export class ChangesList extends React.Component<
     showContextualMenu(items)
   }
 
+  private getPlaceholderMessage(
+    files: ReadonlyArray<WorkingDirectoryFileChange>,
+    singleFileCommit: boolean
+  ) {
+    if (!singleFileCommit) {
+      return 'Summary (required)'
+    }
+
+    const firstFile = files[0]
+    // TODO: can this error? what should happen in that case?
+    const fileName = parse(firstFile.path).base
+
+    switch (firstFile.status) {
+      case AppFileStatus.New:
+        return `Create ${fileName}`
+      case AppFileStatus.Deleted:
+        return `Delete ${fileName}`
+      default:
+        // TODO:
+        // this doesn't feel like a great message for AppFileState.Copied or
+        // AppFileStatus.Renamed but without more insight (and whether this
+        // affects other parts of the flow)
+        return `Update ${fileName}`
+    }
+  }
+
   public render() {
     const fileList = this.props.workingDirectory.files
     const fileCount = fileList.length
@@ -427,15 +453,10 @@ export class ChangesList extends React.Component<
           isCommitting={this.props.isCommitting}
           showCoAuthoredBy={this.props.showCoAuthoredBy}
           coAuthors={this.props.coAuthors}
-          placeholder={
+          placeholder={this.getPlaceholderMessage(
+            filesSelected,
             singleFileCommit
-              ? filesSelected[0].status === AppFileStatus.New
-                ? `Create ${path.parse(filesSelected[0].path).base}`
-                : filesSelected[0].status === AppFileStatus.Deleted
-                  ? `Delete ${path.parse(filesSelected[0].path).base}`
-                  : `Update ${path.parse(filesSelected[0].path).base}`
-              : 'Summary (required)'
-          }
+          )}
           singleFileCommit={singleFileCommit}
         />
       </div>
```

I also had a couple of questions about the new implementation:

 - Will `path.parse` throw an error for paths it cannot handle correctly? Is this something we should be paranoid about? 
 - `AppFileStatus` has other states that could be better described than `Update [file]` - are we likely to stumble upon them? Do we care?

Reply:
Excellent, I had hoped to implement it more cleanly than I had but I didn't think to split it out into it's own function at the time. This should give a bit more wiggle room.

Since we're now using a function, the terseness of the one-liner `path.parse(string).base` may not be necessary. It may be simpler to find the last index of `\` or `/` and substring to everything after it. That way we can always know what to expect from it, and does not require anyone to know what either `.parse()` does or `.base` does.

Additionally, with the second question, both `.Copied` and `.Renamed` feel like good candidates for their own case out of the rest of them.